### PR TITLE
Fix broken links in Napari Code of Conduct

### LINKED ISSUE #5001: "napari Code of Conduct - How to follow up on a report." link broken
## 📚 Documentation
The "[napari Code of Conduct - How to follow up on a report.](https://github.com/napari/napari/blob/main/docs/developers/code_of_conduct_reporting.md)" link at https://napari.org/stable/community/code_of_conduct.html#incident-reporting-resolution-code-of-conduct-enforcement currently gives 404.

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
Fixes https://github.com/napari/napari/issues/5001

Switch to using internal links rather than URLs (and currently broken) links to the internal stuff.
Fix the external link to SciPy Code of Conduct, by using the link from SciPy Community page:
https://scipy.org/community/

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality) N/A
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) N/A
- [ ] This change requires a documentation update N/A

# References
Fixes https://github.com/napari/napari/issues/5001

The external link to SciPy Code of Conduct is from SciPy Community page:
https://scipy.org/community/

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z N/A
- [ ] example: all tests pass with my change N/A
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  N/A

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas N/A
- [ ] I have made corresponding changes to the documentation N/A
- [ ] I have added tests that prove my fix is effective or that my feature works N/A
- [ ] If I included new strings, I have used `trans.` to make them localizable. N/A
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
This is good to review I think—I downloaded the artifact and checked locally.
Do note that in the Numpy Code of Conduct, the link to SciPy was also broken. I did a PR there as well and got the following comment:
https://github.com/numpy/numpy.org/pull/596#discussion_r962144896
I can go either way with the link, I used the SciPy page as the authoritative source, but it's something to consider.

File: docs/community/code_of_conduct.md
Comment (Czaki) [inline docs/community/code_of_conduct.md]: Why this insteado of this: https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html ?

Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: As I mentioned above, I used the link that is used by the SciPy Community page, figuring that it's the authoritative source in this case.
Happy to change if the one you suggest is better.

Comment (alisterburt) [inline docs/community/code_of_conduct.md]:
```suggestion
* [The SciPy Code of Conduct](https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html)
```

**Review by alisterburt — APPROVED** [review]
Very nice, looks good @psobolewskiPhD ! I'm approving - I have a slight preference for the original public facing scipy link so have left reverting that as a suggestion. I will leave this to @Czaki for final decision before merge! 

This will be good to get in for 0.4.17 next week

File: docs/community/code_of_conduct.md
Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: @alisterburt your suggested link is broken—for me at least.

Comment (alisterburt) [reply docs/community/code_of_conduct.md]: fixed, thanks! I prefer the public facing link in this case

Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: OK. The one I used is the one SciPy uses, but 🤷‍♂️

Comment (Czaki) [reply docs/community/code_of_conduct.md]: it looks like they move their API docs to github.io so maybe @psobolewskiPhD link is better.

Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: Again, it's not **my link**, it's from SciPy Community page:
https://scipy.org/community/

Comment (Carreau) [reply docs/community/code_of_conduct.md]: I think this is an oversight from before the docs were moved to docs.scipy.org, as there is a redirect from `scipy.github.io/*` to docs.scipy.org, but not for `scipy.github.io/devdocs`.

I've send https://github.com/scipy/scipy.org/pull/497 to update upstream.

Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: Ok @Carreau so do we need to hold off for the scipy changes so this isn't immediately broken again? Or...

Comment (Carreau) [reply docs/community/code_of_conduct.md]: No, the links will still work, it's just scipy documentation itself that will point to the right location.

Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: ok, so this PR should use the link @czaki pasted? or? Sorry i'm kinda dense (and jet lagged).

**Review by jni — APPROVED** [review]
Thanks @psobolewskiPhD! I think if you use the suggestion from @alisterburt this is ready to go in. Yes the new link is where SciPy will point to in the (near?) future anyway.

File: docs/community/code_of_conduct.md
Comment (psobolewskiPhD) [reply docs/community/code_of_conduct.md]: Used the one from @alisterburt per @jni

**Comment by alisterburt** [comment]
Thanks all! Merging 🚀

**Pull request merged**
